src/mpyl/stages/discovery.py

```diff
@@ -58,27 +69,42 @@ def _to_relevant_changes(
 
 
 def _are_invalidated(
-    project: Project, stage: str, change_history: list[Revision]
+    logger: logging.Logger, project: Project, stage: str, change_history: list[Revision]
 ) -> bool:
     if project.stages.for_stage(stage) is None:
         return False
 
     relevant_changes = _to_relevant_changes(project, stage, change_history)
     return (
-        len(set(filter(lambda c: is_invalidated(project, stage, c), relevant_changes)))
+        len(
+            set(
+                filter(
+                    lambda c: is_invalidated(logger, project, stage, c),
+                    relevant_changes,
+                )
+            )
+        )
         > 0
     )
 
 
 def find_invalidated_projects_for_stage(
-    all_projects: set[Project], stage: str, change_history: list[Revision]
+    logger: logging.Logger,
+    all_projects: set[Project],
+    stage: str,
+    change_history: list[Revision],
 ) -> set[Project]:
     return set(
-        filter(lambda p: _are_invalidated(p, stage, change_history), all_projects)
+        filter(
+            lambda p: _are_invalidated(logger, p, stage, change_history),
+            all_projects,
+        )
     )
 
 
-def find_deploy_set(repo_config: RepoConfig, tag: Optional[str]) -> DeploySet:
+def find_deploy_set(
```

Comment:
I wouldn't apply any changes to find_deploy_set, considering https://github.com/Vandebron/mpyl/pull/320/files#diff-357d00bb7971180759ee4e4284939194e4921adf0a8b4af36736a89e386a2eaa